pkg/cmd/variable/create/create.go

```diff
@@ -0,0 +1,219 @@
+package create
+
+import (
+	"fmt"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmd/variable/shared"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdCreate(f *cmdutil.Factory, runF func(*shared.PostPatchOptions) error) *cobra.Command {
+	opts := &shared.PostPatchOptions{
+		IO:         f.IOStreams,
+		Config:     f.Config,
+		HttpClient: f.HttpClient,
+		Prompter:   f.Prompter,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "create <variable-name>",
+		Short: "Create variables",
+		Long: heredoc.Doc(`
+			Create a value for a variable on one of the following levels:
+			- repository (default): available to Actions runs in a repository
+			- environment: available to Actions runs for a deployment environment in a repository
+			- organization: available to Actions runs within an organization
+
+			Organization variable can optionally be restricted to only be available to
+			specific repositories.
+		`),
+		Example: heredoc.Doc(`
+			# Add variable value for the current repository in an interactive prompt
+			$ gh variable create MYVARIABLE
+
+			# Read variable value from an environment variable
+			$ gh variable create MYVARIABLE --body "$ENV_VALUE"
+
+			# Read variable value from a file
+			$ gh variable create MYVARIABLE < myfile.txt
+
+			# Create variable for a deployment environment in the current repository
+			$ gh variable create MYVARIABLE --env myenvironment
+
+			# Create organization-level variable visible to both public and private repositories
+			$ gh variable create MYVARIABLE --org myOrg --visibility all
+
+			# Create organization-level variable visible to specific repositories
+			$ gh variable create MYVARIABLE --org myOrg --repos repo1,repo2,repo3
+
+		`),
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			if err := cmdutil.MutuallyExclusive("specify only one of `--org` or `--env`", opts.OrgName != "", opts.EnvName != ""); err != nil {
+				return err
+			}
+
+			if err := cmdutil.MutuallyExclusive("specify only one of `--body`", opts.Body != ""); err != nil {
+				return err
+			}
+
+			opts.VariableName = args[0]
```

Comment:
This causes a panic if no variable name is specified.

Reply:
added a check.